base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -2285,26 +2270,31 @@ fn find_orphan_descendant_tips_of<T: BlockchainBackend>(
                 let chain_header = ChainHeader::try_construct(child.header, accum_data).ok_or_else(|| {
                     ChainStorageError::InvalidOperation(format!(
                         "Attempt to create mismatched ChainHeader with hash {}",
-                        child_hash.to_hex()
+                        child_hash,
                     ))
                 })?;
 
                 // Set/overwrite accumulated data for this orphan block
                 let mut txn = DbTransaction::new();
                 txn.set_accumulated_data_for_orphan(chain_header.accumulated_data().clone());
                 db.write(txn)?;
-                let curr_timestamp = chain_header.header().timestamp();
-                let new_prev_timestamps = [&[curr_timestamp], prev_timestamps].concat();
-                let children =
-                    find_orphan_descendant_tips_of(db, chain_header.clone(), &new_prev_timestamps, validator)?;
+                let children = find_orphan_descendant_tips_of(
+                    db,
+                    chain_header,
+                    current_branch_prev_timestamps.clone(),
+                    validator,
+                )?;
                 res.extend(children);
+
+                // Restore the previous timestamps for this level of branching
+                current_branch_prev_timestamps = prev_timestamps.clone();
```

Comment:
why are you resetting completely here?

Reply:
the function is recursive, so it's only resetting to the 'level' - but anyhow, you're correct about not including the timestamp from the current block which removes this line :) I've pushed the changes